// @adyen/api-library/lib/src/typings/legalEntityManagement/document.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Document = void 0;
class Document {
    static getAttributeTypeMap() {
        return Document.attributeTypeMap;
    }
}
exports.Document = Document;
Document.discriminator = undefined;
Document.attributeTypeMap = [
    {
        "name": "attachment",
        "baseName": "attachment",
        "type": "Attachment"
    },
    {
        "name": "attachments",
        "baseName": "attachments",
        "type": "Array<Attachment>"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "expiryDate",
        "baseName": "expiryDate",
        "type": "string"
    },
    {
        "name": "fileName",
        "baseName": "fileName",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "issuerCountry",
        "baseName": "issuerCountry",
        "type": "string"
    },
    {
        "name": "issuerState",
        "baseName": "issuerState",
        "type": "string"
    },
    {
        "name": "number",
        "baseName": "number",
        "type": "string"
    },
    {
        "name": "owner",
        "baseName": "owner",
        "type": "OwnerEntity"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Document.TypeEnum"
    }
];
(function (Document) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BankStatement"] = 'bankStatement'] = "BankStatement";
        TypeEnum[TypeEnum["DriversLicense"] = 'driversLicense'] = "DriversLicense";
        TypeEnum[TypeEnum["IdentityCard"] = 'identityCard'] = "IdentityCard";
        TypeEnum[TypeEnum["NationalIdNumber"] = 'nationalIdNumber'] = "NationalIdNumber";
        TypeEnum[TypeEnum["Passport"] = 'passport'] = "Passport";
        TypeEnum[TypeEnum["ProofOfAddress"] = 'proofOfAddress'] = "ProofOfAddress";
        TypeEnum[TypeEnum["ProofOfNationalIdNumber"] = 'proofOfNationalIdNumber'] = "ProofOfNationalIdNumber";
        TypeEnum[TypeEnum["ProofOfResidency"] = 'proofOfResidency'] = "ProofOfResidency";
        TypeEnum[TypeEnum["RegistrationDocument"] = 'registrationDocument'] = "RegistrationDocument";
        TypeEnum[TypeEnum["VatDocument"] = 'vatDocument'] = "VatDocument";
        TypeEnum[TypeEnum["ProofOfOrganizationTaxInfo"] = 'proofOfOrganizationTaxInfo'] = "ProofOfOrganizationTaxInfo";
        TypeEnum[TypeEnum["ProofOfIndustry"] = 'proofOfIndustry'] = "ProofOfIndustry";
        TypeEnum[TypeEnum["ConstitutionalDocument"] = 'constitutionalDocument'] = "ConstitutionalDocument";
    })(TypeEnum = Document.TypeEnum || (Document.TypeEnum = {}));
})(Document = exports.Document || (exports.Document = {}));
