// @adyen/api-library/lib/src/typings/legalEntityManagement/identificationData.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentificationData = void 0;
class IdentificationData {
    static getAttributeTypeMap() {
        return IdentificationData.attributeTypeMap;
    }
}
exports.IdentificationData = IdentificationData;
IdentificationData.discriminator = undefined;
IdentificationData.attributeTypeMap = [
    {
        "name": "expiryDate",
        "baseName": "expiryDate",
        "type": "string"
    },
    {
        "name": "issuerCountry",
        "baseName": "issuerCountry",
        "type": "string"
    },
    {
        "name": "issuerState",
        "baseName": "issuerState",
        "type": "string"
    },
    {
        "name": "nationalIdExempt",
        "baseName": "nationalIdExempt",
        "type": "boolean"
    },
    {
        "name": "number",
        "baseName": "number",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "IdentificationData.TypeEnum"
    }
];
(function (IdentificationData) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["BankStatement"] = 'bankStatement'] = "BankStatement";
        TypeEnum[TypeEnum["DriversLicense"] = 'driversLicense'] = "DriversLicense";
        TypeEnum[TypeEnum["IdentityCard"] = 'identityCard'] = "IdentityCard";
        TypeEnum[TypeEnum["NationalIdNumber"] = 'nationalIdNumber'] = "NationalIdNumber";
        TypeEnum[TypeEnum["Passport"] = 'passport'] = "Passport";
        TypeEnum[TypeEnum["ProofOfAddress"] = 'proofOfAddress'] = "ProofOfAddress";
        TypeEnum[TypeEnum["ProofOfNationalIdNumber"] = 'proofOfNationalIdNumber'] = "ProofOfNationalIdNumber";
        TypeEnum[TypeEnum["ProofOfResidency"] = 'proofOfResidency'] = "ProofOfResidency";
        TypeEnum[TypeEnum["RegistrationDocument"] = 'registrationDocument'] = "RegistrationDocument";
        TypeEnum[TypeEnum["VatDocument"] = 'vatDocument'] = "VatDocument";
        TypeEnum[TypeEnum["ProofOfOrganizationTaxInfo"] = 'proofOfOrganizationTaxInfo'] = "ProofOfOrganizationTaxInfo";
        TypeEnum[TypeEnum["ProofOfIndustry"] = 'proofOfIndustry'] = "ProofOfIndustry";
        TypeEnum[TypeEnum["ConstitutionalDocument"] = 'constitutionalDocument'] = "ConstitutionalDocument";
    })(TypeEnum = IdentificationData.TypeEnum || (IdentificationData.TypeEnum = {}));
})(IdentificationData = exports.IdentificationData || (exports.IdentificationData = {}));
