// @adyen/api-library/lib/src/typings/legalEntityManagement/legalEntity.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntity = void 0;
class LegalEntity {
    static getAttributeTypeMap() {
        return LegalEntity.attributeTypeMap;
    }
}
exports.LegalEntity = LegalEntity;
LegalEntity.discriminator = undefined;
LegalEntity.attributeTypeMap = [
    {
        "name": "documentDetails",
        "baseName": "documentDetails",
        "type": "Array<DocumentReference>"
    },
    {
        "name": "documents",
        "baseName": "documents",
        "type": "Array<EntityReference>"
    },
    {
        "name": "entityAssociations",
        "baseName": "entityAssociations",
        "type": "Array<LegalEntityAssociation>"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "individual",
        "baseName": "individual",
        "type": "Individual"
    },
    {
        "name": "organization",
        "baseName": "organization",
        "type": "Organization"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "soleProprietorship",
        "baseName": "soleProprietorship",
        "type": "SoleProprietorship"
    },
    {
        "name": "transferInstruments",
        "baseName": "transferInstruments",
        "type": "Array<EntityReference>"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalEntity.TypeEnum"
    }
];
(function (LegalEntity) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Individual"] = 'individual'] = "Individual";
        TypeEnum[TypeEnum["Organization"] = 'organization'] = "Organization";
        TypeEnum[TypeEnum["SoleProprietorship"] = 'soleProprietorship'] = "SoleProprietorship";
        TypeEnum[TypeEnum["Trust"] = 'trust'] = "Trust";
        TypeEnum[TypeEnum["UnincorporatedPartnership"] = 'unincorporatedPartnership'] = "UnincorporatedPartnership";
    })(TypeEnum = LegalEntity.TypeEnum || (LegalEntity.TypeEnum = {}));
})(LegalEntity = exports.LegalEntity || (exports.LegalEntity = {}));
