// @adyen/api-library/lib/src/typings/legalEntityManagement/legalEntityInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntityInfo = void 0;
class LegalEntityInfo {
    static getAttributeTypeMap() {
        return LegalEntityInfo.attributeTypeMap;
    }
}
exports.LegalEntityInfo = LegalEntityInfo;
LegalEntityInfo.discriminator = undefined;
LegalEntityInfo.attributeTypeMap = [
    {
        "name": "entityAssociations",
        "baseName": "entityAssociations",
        "type": "Array<LegalEntityAssociation>"
    },
    {
        "name": "individual",
        "baseName": "individual",
        "type": "Individual"
    },
    {
        "name": "organization",
        "baseName": "organization",
        "type": "Organization"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "soleProprietorship",
        "baseName": "soleProprietorship",
        "type": "SoleProprietorship"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalEntityInfo.TypeEnum"
    }
];
(function (LegalEntityInfo) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Individual"] = 'individual'] = "Individual";
        TypeEnum[TypeEnum["Organization"] = 'organization'] = "Organization";
        TypeEnum[TypeEnum["SoleProprietorship"] = 'soleProprietorship'] = "SoleProprietorship";
        TypeEnum[TypeEnum["Trust"] = 'trust'] = "Trust";
        TypeEnum[TypeEnum["UnincorporatedPartnership"] = 'unincorporatedPartnership'] = "UnincorporatedPartnership";
    })(TypeEnum = LegalEntityInfo.TypeEnum || (LegalEntityInfo.TypeEnum = {}));
})(LegalEntityInfo = exports.LegalEntityInfo || (exports.LegalEntityInfo = {}));
