// @adyen/api-library/lib/src/typings/legalEntityManagement/organization.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Organization = void 0;
class Organization {
    static getAttributeTypeMap() {
        return Organization.attributeTypeMap;
    }
}
exports.Organization = Organization;
Organization.discriminator = undefined;
Organization.attributeTypeMap = [
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "doingBusinessAs",
        "baseName": "doingBusinessAs",
        "type": "string"
    },
    {
        "name": "email",
        "baseName": "email",
        "type": "string"
    },
    {
        "name": "legalName",
        "baseName": "legalName",
        "type": "string"
    },
    {
        "name": "phone",
        "baseName": "phone",
        "type": "PhoneNumber"
    },
    {
        "name": "principalPlaceOfBusiness",
        "baseName": "principalPlaceOfBusiness",
        "type": "Address"
    },
    {
        "name": "registeredAddress",
        "baseName": "registeredAddress",
        "type": "Address"
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string"
    },
    {
        "name": "stockData",
        "baseName": "stockData",
        "type": "StockData"
    },
    {
        "name": "taxInformation",
        "baseName": "taxInformation",
        "type": "Array<TaxInformation>"
    },
    {
        "name": "taxReportingClassification",
        "baseName": "taxReportingClassification",
        "type": "TaxReportingClassification"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Organization.TypeEnum"
    },
    {
        "name": "vatAbsenceReason",
        "baseName": "vatAbsenceReason",
        "type": "Organization.VatAbsenceReasonEnum"
    },
    {
        "name": "vatNumber",
        "baseName": "vatNumber",
        "type": "string"
    },
    {
        "name": "webData",
        "baseName": "webData",
        "type": "WebData"
    }
];
(function (Organization) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["AssociationIncorporated"] = 'associationIncorporated'] = "AssociationIncorporated";
        TypeEnum[TypeEnum["GovernmentalOrganization"] = 'governmentalOrganization'] = "GovernmentalOrganization";
        TypeEnum[TypeEnum["ListedPublicCompany"] = 'listedPublicCompany'] = "ListedPublicCompany";
        TypeEnum[TypeEnum["NonProfit"] = 'nonProfit'] = "NonProfit";
        TypeEnum[TypeEnum["PartnershipIncorporated"] = 'partnershipIncorporated'] = "PartnershipIncorporated";
        TypeEnum[TypeEnum["PrivateCompany"] = 'privateCompany'] = "PrivateCompany";
    })(TypeEnum = Organization.TypeEnum || (Organization.TypeEnum = {}));
    let VatAbsenceReasonEnum;
    (function (VatAbsenceReasonEnum) {
        VatAbsenceReasonEnum[VatAbsenceReasonEnum["IndustryExemption"] = 'industryExemption'] = "IndustryExemption";
        VatAbsenceReasonEnum[VatAbsenceReasonEnum["BelowTaxThreshold"] = 'belowTaxThreshold'] = "BelowTaxThreshold";
    })(VatAbsenceReasonEnum = Organization.VatAbsenceReasonEnum || (Organization.VatAbsenceReasonEnum = {}));
})(Organization = exports.Organization || (exports.Organization = {}));
