// @adyen/api-library/lib/src/typings/legalEntityManagement/taxReportingClassification.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxReportingClassification = void 0;
class TaxReportingClassification {
    static getAttributeTypeMap() {
        return TaxReportingClassification.attributeTypeMap;
    }
}
exports.TaxReportingClassification = TaxReportingClassification;
TaxReportingClassification.discriminator = undefined;
TaxReportingClassification.attributeTypeMap = [
    {
        "name": "businessType",
        "baseName": "businessType",
        "type": "TaxReportingClassification.BusinessTypeEnum"
    },
    {
        "name": "financialInstitutionNumber",
        "baseName": "financialInstitutionNumber",
        "type": "string"
    },
    {
        "name": "mainSourceOfIncome",
        "baseName": "mainSourceOfIncome",
        "type": "TaxReportingClassification.MainSourceOfIncomeEnum"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TaxReportingClassification.TypeEnum"
    }
];
(function (TaxReportingClassification) {
    let BusinessTypeEnum;
    (function (BusinessTypeEnum) {
        BusinessTypeEnum[BusinessTypeEnum["Other"] = 'other'] = "Other";
        BusinessTypeEnum[BusinessTypeEnum["ListedPublicCompany"] = 'listedPublicCompany'] = "ListedPublicCompany";
        BusinessTypeEnum[BusinessTypeEnum["SubsidiaryOfListedPublicCompany"] = 'subsidiaryOfListedPublicCompany'] = "SubsidiaryOfListedPublicCompany";
        BusinessTypeEnum[BusinessTypeEnum["GovernmentalOrganization"] = 'governmentalOrganization'] = "GovernmentalOrganization";
        BusinessTypeEnum[BusinessTypeEnum["InternationalOrganization"] = 'internationalOrganization'] = "InternationalOrganization";
        BusinessTypeEnum[BusinessTypeEnum["FinancialInstitution"] = 'financialInstitution.'] = "FinancialInstitution";
    })(BusinessTypeEnum = TaxReportingClassification.BusinessTypeEnum || (TaxReportingClassification.BusinessTypeEnum = {}));
    let MainSourceOfIncomeEnum;
    (function (MainSourceOfIncomeEnum) {
        MainSourceOfIncomeEnum[MainSourceOfIncomeEnum["BusinessOperation"] = 'businessOperation'] = "BusinessOperation";
        MainSourceOfIncomeEnum[MainSourceOfIncomeEnum["RealEstateSales"] = 'realEstateSales'] = "RealEstateSales";
        MainSourceOfIncomeEnum[MainSourceOfIncomeEnum["InvestmentInterestOrRoyalty"] = 'investmentInterestOrRoyalty'] = "InvestmentInterestOrRoyalty";
        MainSourceOfIncomeEnum[MainSourceOfIncomeEnum["PropertyRental"] = 'propertyRental'] = "PropertyRental";
        MainSourceOfIncomeEnum[MainSourceOfIncomeEnum["Other"] = 'other'] = "Other";
    })(MainSourceOfIncomeEnum = TaxReportingClassification.MainSourceOfIncomeEnum || (TaxReportingClassification.MainSourceOfIncomeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["NonFinancialNonReportable"] = 'nonFinancialNonReportable'] = "NonFinancialNonReportable";
        TypeEnum[TypeEnum["FinancialNonReportable"] = 'financialNonReportable'] = "FinancialNonReportable";
        TypeEnum[TypeEnum["NonFinancialActive"] = 'nonFinancialActive'] = "NonFinancialActive";
        TypeEnum[TypeEnum["NonFinancialPassive"] = 'nonFinancialPassive'] = "NonFinancialPassive";
    })(TypeEnum = TaxReportingClassification.TypeEnum || (TaxReportingClassification.TypeEnum = {}));
})(TaxReportingClassification = exports.TaxReportingClassification || (exports.TaxReportingClassification = {}));
