// @adyen/api-library/lib/src/typings/balancePlatform/cronSweepSchedule.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CronSweepSchedule = void 0;
class CronSweepSchedule {
    static getAttributeTypeMap() {
        return CronSweepSchedule.attributeTypeMap;
    }
}
exports.CronSweepSchedule = CronSweepSchedule;
CronSweepSchedule.discriminator = undefined;
CronSweepSchedule.attributeTypeMap = [
    {
        "name": "cronExpression",
        "baseName": "cronExpression",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CronSweepSchedule.TypeEnum"
    }
];
(function (CronSweepSchedule) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["Daily"] = 'daily'] = "Daily";
        TypeEnum[TypeEnum["Weekly"] = 'weekly'] = "Weekly";
        TypeEnum[TypeEnum["Monthly"] = 'monthly'] = "Monthly";
        TypeEnum[TypeEnum["Balance"] = 'balance'] = "Balance";
        TypeEnum[TypeEnum["Cron"] = 'cron'] = "Cron";
    })(TypeEnum = CronSweepSchedule.TypeEnum || (CronSweepSchedule.TypeEnum = {}));
})(CronSweepSchedule = exports.CronSweepSchedule || (exports.CronSweepSchedule = {}));
