// @adyen/api-library/lib/src/typings/management/paymentMethod.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethod = void 0;
class PaymentMethod {
    static getAttributeTypeMap() {
        return PaymentMethod.attributeTypeMap;
    }
}
exports.PaymentMethod = PaymentMethod;
PaymentMethod.discriminator = undefined;
PaymentMethod.attributeTypeMap = [
    {
        "name": "allowed",
        "baseName": "allowed",
        "type": "boolean"
    },
    {
        "name": "applePay",
        "baseName": "applePay",
        "type": "ApplePayInfo"
    },
    {
        "name": "bcmc",
        "baseName": "bcmc",
        "type": "BcmcInfo"
    },
    {
        "name": "businessLineId",
        "baseName": "businessLineId",
        "type": "string"
    },
    {
        "name": "cartesBancaires",
        "baseName": "cartesBancaires",
        "type": "CartesBancairesInfo"
    },
    {
        "name": "countries",
        "baseName": "countries",
        "type": "Array<string>"
    },
    {
        "name": "currencies",
        "baseName": "currencies",
        "type": "Array<string>"
    },
    {
        "name": "enabled",
        "baseName": "enabled",
        "type": "boolean"
    },
    {
        "name": "giroPay",
        "baseName": "giroPay",
        "type": "GiroPayInfo"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "klarna",
        "baseName": "klarna",
        "type": "KlarnaInfo"
    },
    {
        "name": "paypal",
        "baseName": "paypal",
        "type": "PayPalInfo"
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "string"
    },
    {
        "name": "sofort",
        "baseName": "sofort",
        "type": "SofortInfo"
    },
    {
        "name": "storeId",
        "baseName": "storeId",
        "type": "string"
    },
    {
        "name": "swish",
        "baseName": "swish",
        "type": "SwishInfo"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string"
    },
    {
        "name": "verificationStatus",
        "baseName": "verificationStatus",
        "type": "PaymentMethod.VerificationStatusEnum"
    }
];
(function (PaymentMethod) {
    let VerificationStatusEnum;
    (function (VerificationStatusEnum) {
        VerificationStatusEnum[VerificationStatusEnum["Valid"] = 'valid'] = "Valid";
        VerificationStatusEnum[VerificationStatusEnum["Pending"] = 'pending'] = "Pending";
        VerificationStatusEnum[VerificationStatusEnum["Invalid"] = 'invalid'] = "Invalid";
        VerificationStatusEnum[VerificationStatusEnum["Rejected"] = 'rejected'] = "Rejected";
    })(VerificationStatusEnum = PaymentMethod.VerificationStatusEnum || (PaymentMethod.VerificationStatusEnum = {}));
})(PaymentMethod = exports.PaymentMethod || (exports.PaymentMethod = {}));
