// @adyen/api-library/lib/src/typings/transfer/transferInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferInfo = void 0;
class TransferInfo {
    static getAttributeTypeMap() {
        return TransferInfo.attributeTypeMap;
    }
}
exports.TransferInfo = TransferInfo;
TransferInfo.discriminator = undefined;
TransferInfo.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "TransferInfo.CategoryEnum"
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "CounterpartyInfoV3"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "paymentInstrumentId",
        "baseName": "paymentInstrumentId",
        "type": "string"
    },
    {
        "name": "priority",
        "baseName": "priority",
        "type": "TransferInfo.PriorityEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "referenceForBeneficiary",
        "baseName": "referenceForBeneficiary",
        "type": "string"
    }
];
(function (TransferInfo) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum[CategoryEnum["Bank"] = 'bank'] = "Bank";
        CategoryEnum[CategoryEnum["Internal"] = 'internal'] = "Internal";
    })(CategoryEnum = TransferInfo.CategoryEnum || (TransferInfo.CategoryEnum = {}));
    let PriorityEnum;
    (function (PriorityEnum) {
        PriorityEnum[PriorityEnum["Fast"] = 'fast'] = "Fast";
        PriorityEnum[PriorityEnum["Instant"] = 'instant'] = "Instant";
        PriorityEnum[PriorityEnum["Regular"] = 'regular'] = "Regular";
        PriorityEnum[PriorityEnum["Wire"] = 'wire'] = "Wire";
    })(PriorityEnum = TransferInfo.PriorityEnum || (TransferInfo.PriorityEnum = {}));
})(TransferInfo = exports.TransferInfo || (exports.TransferInfo = {}));
