// @adyen/api-library/lib/src/typings/transfer/transaction.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transaction = void 0;
class Transaction {
    static getAttributeTypeMap() {
        return Transaction.attributeTypeMap;
    }
}
exports.Transaction = Transaction;
Transaction.discriminator = undefined;
Transaction.attributeTypeMap = [
    {
        "name": "accountHolderId",
        "baseName": "accountHolderId",
        "type": "string"
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "balancePlatform",
        "baseName": "balancePlatform",
        "type": "string"
    },
    {
        "name": "bookingDate",
        "baseName": "bookingDate",
        "type": "Date"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "Transaction.CategoryEnum"
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "CounterpartyV3"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "instructedAmount",
        "baseName": "instructedAmount",
        "type": "Amount"
    },
    {
        "name": "paymentInstrumentId",
        "baseName": "paymentInstrumentId",
        "type": "string"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "referenceForBeneficiary",
        "baseName": "referenceForBeneficiary",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "Transaction.StatusEnum"
    },
    {
        "name": "transferId",
        "baseName": "transferId",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Transaction.TypeEnum"
    },
    {
        "name": "valueDate",
        "baseName": "valueDate",
        "type": "Date"
    }
];
(function (Transaction) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum[CategoryEnum["Bank"] = 'bank'] = "Bank";
        CategoryEnum[CategoryEnum["Grants"] = 'grants'] = "Grants";
        CategoryEnum[CategoryEnum["Internal"] = 'internal'] = "Internal";
        CategoryEnum[CategoryEnum["IssuedCard"] = 'issuedCard'] = "IssuedCard";
        CategoryEnum[CategoryEnum["PlatformPayment"] = 'platformPayment'] = "PlatformPayment";
    })(CategoryEnum = Transaction.CategoryEnum || (Transaction.CategoryEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Booked"] = 'booked'] = "Booked";
        StatusEnum[StatusEnum["Pending"] = 'pending'] = "Pending";
    })(StatusEnum = Transaction.StatusEnum || (Transaction.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["AtmWithdrawal"] = 'atmWithdrawal'] = "AtmWithdrawal";
        TypeEnum[TypeEnum["AtmWithdrawalReversal"] = 'atmWithdrawalReversal'] = "AtmWithdrawalReversal";
        TypeEnum[TypeEnum["BalanceAdjustment"] = 'balanceAdjustment'] = "BalanceAdjustment";
        TypeEnum[TypeEnum["BankTransfer"] = 'bankTransfer'] = "BankTransfer";
        TypeEnum[TypeEnum["Capture"] = 'capture'] = "Capture";
        TypeEnum[TypeEnum["CaptureReversal"] = 'captureReversal'] = "CaptureReversal";
        TypeEnum[TypeEnum["Chargeback"] = 'chargeback'] = "Chargeback";
        TypeEnum[TypeEnum["ChargebackReversal"] = 'chargebackReversal'] = "ChargebackReversal";
        TypeEnum[TypeEnum["DepositCorrection"] = 'depositCorrection'] = "DepositCorrection";
        TypeEnum[TypeEnum["Fee"] = 'fee'] = "Fee";
        TypeEnum[TypeEnum["Grant"] = 'grant'] = "Grant";
        TypeEnum[TypeEnum["Installment"] = 'installment'] = "Installment";
        TypeEnum[TypeEnum["InstallmentReversal"] = 'installmentReversal'] = "InstallmentReversal";
        TypeEnum[TypeEnum["InternalTransfer"] = 'internalTransfer'] = "InternalTransfer";
        TypeEnum[TypeEnum["InvoiceDeduction"] = 'invoiceDeduction'] = "InvoiceDeduction";
        TypeEnum[TypeEnum["Leftover"] = 'leftover'] = "Leftover";
        TypeEnum[TypeEnum["ManualCorrection"] = 'manualCorrection'] = "ManualCorrection";
        TypeEnum[TypeEnum["MiscCost"] = 'miscCost'] = "MiscCost";
        TypeEnum[TypeEnum["Payment"] = 'payment'] = "Payment";
        TypeEnum[TypeEnum["PaymentCost"] = 'paymentCost'] = "PaymentCost";
        TypeEnum[TypeEnum["Refund"] = 'refund'] = "Refund";
        TypeEnum[TypeEnum["RefundReversal"] = 'refundReversal'] = "RefundReversal";
        TypeEnum[TypeEnum["Repayment"] = 'repayment'] = "Repayment";
        TypeEnum[TypeEnum["ReserveAdjustment"] = 'reserveAdjustment'] = "ReserveAdjustment";
        TypeEnum[TypeEnum["SecondChargeback"] = 'secondChargeback'] = "SecondChargeback";
    })(TypeEnum = Transaction.TypeEnum || (Transaction.TypeEnum = {}));
})(Transaction = exports.Transaction || (exports.Transaction = {}));
