// @adyen/api-library/lib/src/typings/transfer/transfer.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transfer = void 0;
class Transfer {
    static getAttributeTypeMap() {
        return Transfer.attributeTypeMap;
    }
}
exports.Transfer = Transfer;
Transfer.discriminator = undefined;
Transfer.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount"
    },
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "Transfer.CategoryEnum"
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "CounterpartyV3"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "direction",
        "baseName": "direction",
        "type": "Transfer.DirectionEnum"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "paymentInstrumentId",
        "baseName": "paymentInstrumentId",
        "type": "string"
    },
    {
        "name": "priority",
        "baseName": "priority",
        "type": "Transfer.PriorityEnum"
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "Transfer.ReasonEnum"
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string"
    },
    {
        "name": "referenceForBeneficiary",
        "baseName": "referenceForBeneficiary",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "Transfer.StatusEnum"
    }
];
(function (Transfer) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum[CategoryEnum["Bank"] = 'bank'] = "Bank";
        CategoryEnum[CategoryEnum["Internal"] = 'internal'] = "Internal";
    })(CategoryEnum = Transfer.CategoryEnum || (Transfer.CategoryEnum = {}));
    let DirectionEnum;
    (function (DirectionEnum) {
        DirectionEnum[DirectionEnum["Incoming"] = 'incoming'] = "Incoming";
        DirectionEnum[DirectionEnum["Outgoing"] = 'outgoing'] = "Outgoing";
    })(DirectionEnum = Transfer.DirectionEnum || (Transfer.DirectionEnum = {}));
    let PriorityEnum;
    (function (PriorityEnum) {
        PriorityEnum[PriorityEnum["Fast"] = 'fast'] = "Fast";
        PriorityEnum[PriorityEnum["Instant"] = 'instant'] = "Instant";
        PriorityEnum[PriorityEnum["Regular"] = 'regular'] = "Regular";
        PriorityEnum[PriorityEnum["Wire"] = 'wire'] = "Wire";
    })(PriorityEnum = Transfer.PriorityEnum || (Transfer.PriorityEnum = {}));
    let ReasonEnum;
    (function (ReasonEnum) {
        ReasonEnum[ReasonEnum["Approved"] = 'approved'] = "Approved";
        ReasonEnum[ReasonEnum["Error"] = 'error'] = "Error";
        ReasonEnum[ReasonEnum["NotEnoughBalance"] = 'notEnoughBalance'] = "NotEnoughBalance";
        ReasonEnum[ReasonEnum["Unknown"] = 'unknown'] = "Unknown";
    })(ReasonEnum = Transfer.ReasonEnum || (Transfer.ReasonEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["AtmWithdrawal"] = 'atmWithdrawal'] = "AtmWithdrawal";
        StatusEnum[StatusEnum["AtmWithdrawalReversalPending"] = 'atmWithdrawalReversalPending'] = "AtmWithdrawalReversalPending";
        StatusEnum[StatusEnum["AtmWithdrawalReversed"] = 'atmWithdrawalReversed'] = "AtmWithdrawalReversed";
        StatusEnum[StatusEnum["Authorised"] = 'authorised'] = "Authorised";
        StatusEnum[StatusEnum["BankTransfer"] = 'bankTransfer'] = "BankTransfer";
        StatusEnum[StatusEnum["BankTransferPending"] = 'bankTransferPending'] = "BankTransferPending";
        StatusEnum[StatusEnum["Booked"] = 'booked'] = "Booked";
        StatusEnum[StatusEnum["BookingPending"] = 'bookingPending'] = "BookingPending";
        StatusEnum[StatusEnum["Cancelled"] = 'cancelled'] = "Cancelled";
        StatusEnum[StatusEnum["CapturePending"] = 'capturePending'] = "CapturePending";
        StatusEnum[StatusEnum["CaptureReversalPending"] = 'captureReversalPending'] = "CaptureReversalPending";
        StatusEnum[StatusEnum["CaptureReversed"] = 'captureReversed'] = "CaptureReversed";
        StatusEnum[StatusEnum["Captured"] = 'captured'] = "Captured";
        StatusEnum[StatusEnum["Chargeback"] = 'chargeback'] = "Chargeback";
        StatusEnum[StatusEnum["ChargebackPending"] = 'chargebackPending'] = "ChargebackPending";
        StatusEnum[StatusEnum["ChargebackReversalPending"] = 'chargebackReversalPending'] = "ChargebackReversalPending";
        StatusEnum[StatusEnum["ChargebackReversed"] = 'chargebackReversed'] = "ChargebackReversed";
        StatusEnum[StatusEnum["DepositCorrection"] = 'depositCorrection'] = "DepositCorrection";
        StatusEnum[StatusEnum["DepositCorrectionPending"] = 'depositCorrectionPending'] = "DepositCorrectionPending";
        StatusEnum[StatusEnum["Dispute"] = 'dispute'] = "Dispute";
        StatusEnum[StatusEnum["DisputeClosed"] = 'disputeClosed'] = "DisputeClosed";
        StatusEnum[StatusEnum["DisputeExpired"] = 'disputeExpired'] = "DisputeExpired";
        StatusEnum[StatusEnum["DisputeNeedsReview"] = 'disputeNeedsReview'] = "DisputeNeedsReview";
        StatusEnum[StatusEnum["Error"] = 'error'] = "Error";
        StatusEnum[StatusEnum["Expired"] = 'expired'] = "Expired";
        StatusEnum[StatusEnum["Failed"] = 'failed'] = "Failed";
        StatusEnum[StatusEnum["Fee"] = 'fee'] = "Fee";
        StatusEnum[StatusEnum["FeePending"] = 'feePending'] = "FeePending";
        StatusEnum[StatusEnum["InternalTransfer"] = 'internalTransfer'] = "InternalTransfer";
        StatusEnum[StatusEnum["InternalTransferPending"] = 'internalTransferPending'] = "InternalTransferPending";
        StatusEnum[StatusEnum["InvoiceDeduction"] = 'invoiceDeduction'] = "InvoiceDeduction";
        StatusEnum[StatusEnum["InvoiceDeductionPending"] = 'invoiceDeductionPending'] = "InvoiceDeductionPending";
        StatusEnum[StatusEnum["ManualCorrectionPending"] = 'manualCorrectionPending'] = "ManualCorrectionPending";
        StatusEnum[StatusEnum["ManuallyCorrected"] = 'manuallyCorrected'] = "ManuallyCorrected";
        StatusEnum[StatusEnum["MatchedStatement"] = 'matchedStatement'] = "MatchedStatement";
        StatusEnum[StatusEnum["MatchedStatementPending"] = 'matchedStatementPending'] = "MatchedStatementPending";
        StatusEnum[StatusEnum["MerchantPayin"] = 'merchantPayin'] = "MerchantPayin";
        StatusEnum[StatusEnum["MerchantPayinPending"] = 'merchantPayinPending'] = "MerchantPayinPending";
        StatusEnum[StatusEnum["MerchantPayinReversed"] = 'merchantPayinReversed'] = "MerchantPayinReversed";
        StatusEnum[StatusEnum["MerchantPayinReversedPending"] = 'merchantPayinReversedPending'] = "MerchantPayinReversedPending";
        StatusEnum[StatusEnum["MiscCost"] = 'miscCost'] = "MiscCost";
        StatusEnum[StatusEnum["MiscCostPending"] = 'miscCostPending'] = "MiscCostPending";
        StatusEnum[StatusEnum["PaymentCost"] = 'paymentCost'] = "PaymentCost";
        StatusEnum[StatusEnum["PaymentCostPending"] = 'paymentCostPending'] = "PaymentCostPending";
        StatusEnum[StatusEnum["Received"] = 'received'] = "Received";
        StatusEnum[StatusEnum["RefundPending"] = 'refundPending'] = "RefundPending";
        StatusEnum[StatusEnum["RefundReversalPending"] = 'refundReversalPending'] = "RefundReversalPending";
        StatusEnum[StatusEnum["RefundReversed"] = 'refundReversed'] = "RefundReversed";
        StatusEnum[StatusEnum["Refunded"] = 'refunded'] = "Refunded";
        StatusEnum[StatusEnum["Refused"] = 'refused'] = "Refused";
        StatusEnum[StatusEnum["ReserveAdjustment"] = 'reserveAdjustment'] = "ReserveAdjustment";
        StatusEnum[StatusEnum["ReserveAdjustmentPending"] = 'reserveAdjustmentPending'] = "ReserveAdjustmentPending";
        StatusEnum[StatusEnum["SecondChargeback"] = 'secondChargeback'] = "SecondChargeback";
        StatusEnum[StatusEnum["SecondChargebackPending"] = 'secondChargebackPending'] = "SecondChargebackPending";
        StatusEnum[StatusEnum["Undefined"] = 'undefined'] = "Undefined";
    })(StatusEnum = Transfer.StatusEnum || (Transfer.StatusEnum = {}));
})(Transfer = exports.Transfer || (exports.Transfer = {}));
