// @adyen/api-library/lib/src/typings/platformsAccount/payoutScheduleResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v6
 * Contact: developer-experience@adyen.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutScheduleResponse = void 0;
class PayoutScheduleResponse {
    static getAttributeTypeMap() {
        return PayoutScheduleResponse.attributeTypeMap;
    }
}
exports.PayoutScheduleResponse = PayoutScheduleResponse;
PayoutScheduleResponse.discriminator = undefined;
PayoutScheduleResponse.attributeTypeMap = [
    {
        "name": "nextScheduledPayout",
        "baseName": "nextScheduledPayout",
        "type": "Date"
    },
    {
        "name": "schedule",
        "baseName": "schedule",
        "type": "PayoutScheduleResponse.ScheduleEnum"
    }
];
(function (PayoutScheduleResponse) {
    let ScheduleEnum;
    (function (ScheduleEnum) {
        ScheduleEnum[ScheduleEnum["BiweeklyOn1StAnd15ThAtMidnight"] = 'BIWEEKLY_ON_1ST_AND_15TH_AT_MIDNIGHT'] = "BiweeklyOn1StAnd15ThAtMidnight";
        ScheduleEnum[ScheduleEnum["Daily"] = 'DAILY'] = "Daily";
        ScheduleEnum[ScheduleEnum["DailyAu"] = 'DAILY_AU'] = "DailyAu";
        ScheduleEnum[ScheduleEnum["DailyEu"] = 'DAILY_EU'] = "DailyEu";
        ScheduleEnum[ScheduleEnum["DailySg"] = 'DAILY_SG'] = "DailySg";
        ScheduleEnum[ScheduleEnum["DailyUs"] = 'DAILY_US'] = "DailyUs";
        ScheduleEnum[ScheduleEnum["Hold"] = 'HOLD'] = "Hold";
        ScheduleEnum[ScheduleEnum["Monthly"] = 'MONTHLY'] = "Monthly";
        ScheduleEnum[ScheduleEnum["Weekly"] = 'WEEKLY'] = "Weekly";
        ScheduleEnum[ScheduleEnum["WeeklyOnTueFriMidnight"] = 'WEEKLY_ON_TUE_FRI_MIDNIGHT'] = "WeeklyOnTueFriMidnight";
    })(ScheduleEnum = PayoutScheduleResponse.ScheduleEnum || (PayoutScheduleResponse.ScheduleEnum = {}));
})(PayoutScheduleResponse = exports.PayoutScheduleResponse || (exports.PayoutScheduleResponse = {}));
